### PR TITLE
Update tektoncd/pipeline to v0.47.0

### PR DESCRIPTION
**Pull request by lcarva**
# Changes

Update tektoncd/pipeline to v0.47.0. This is a manual update due to the required code changes.

# Submitter Checklist

As the author of this PR, please check off the items in this checklist:

- [x] Has [Docs](https://github.com/tektoncd/community/blob/main/standards.md#docs) included if any changes are user facing
- [x] Has [Tests](https://github.com/tektoncd/community/blob/main/standards.md#tests) included if any functionality added or changed
- [x] Follows the [commit message standard](https://github.com/tektoncd/community/blob/main/standards.md#commits)
- [x] Meets the [Tekton contributor standards](https://github.com/tektoncd/community/blob/main/standards.md) (including
  functionality, content, code)
- [x] Release notes block below has been updated with any user facing changes (API changes, bug fixes, changes requiring upgrade notices or deprecation warnings)
- [x] Release notes contains the string "action required" if the change requires additional action from users switching to the new release

# Release Notes

``` release-note
NONE
```

### CODE REVIEW

**Comment by lcarva** [comment]
@wlynch, @chuangw6, @chitrangpatel PTAL :pray:

File: pkg/artifacts/signable.go [hunk 1]
```diff
@@ -163,7 +162,7 @@ func (oa *OCIArtifact) ExtractObjects(ctx context.Context, obj objects.TektonObj
 		imageResourceNames := map[string]*image{}
 		if tr.Status.TaskSpec != nil && tr.Status.TaskSpec.Resources != nil {
 			for _, output := range tr.Status.TaskSpec.Resources.Outputs {
-				if output.Type == v1alpha1.PipelineResourceTypeImage {
```
Comment (wlynch) [inline pkg/artifacts/signable.go:166]: 😢 Pipelines probably shouldn't have removed these.

But if we have to back-port it let's break these into our own consts (perhaps under `internal/backport`?)

**Review by wlynch — CHANGES_REQUESTED** [review]
LGTM! Just one small nit.

File: pkg/artifacts/signable.go (hunk 1 above)
Comment (lcarva) [reply pkg/artifacts/signable.go:166]: Good call! Done.

**Review by wlynch — APPROVED** [review]

**Pull request merged**